TMS.Net07.Homework.DaysOfWeek/Master/Program.cs

```diff
@@ -0,0 +1,171 @@
+                            break;
+                        case 2:
+                            if (year%4 == 0) //leap year
+                            {
+                                if (day > 0 && day < 30)
+                                    return true;
+                            }
+                            else
+                            {
+                                if (day > 0 && day < 29)
+                                    return true;
+                            }
+                            break;
+                    }
+                }
+            }
+            return false;
+        }
+        static int CalculateAmountOfDays(int day, int month, int year)
+        {
+            int amountPastYears = (365 * (year - 1)) + ((year - 1) / 4);
+            int amountThisYear = 0;
+            switch (month - 1)
+            {
+                case 0:
+                    amountThisYear = day;
+                    break;
+                case 1:
+                    amountThisYear = 31 + day;
+                    break;
+                case 2:
+                    amountThisYear = 59 + day;
+                    if (year%4 == 0)
+                        amountThisYear++;
+                    break;
+                case 3:
+                    amountThisYear = 90 + day;
+                    if (year % 4 == 0)
+                        amountThisYear++;
+                    break;
+                case 4:
+                    amountThisYear = 120 + day;
+                    if (year % 4 == 0)
+                        amountThisYear++;
+                    break;
+                case 5:
+                    amountThisYear = 151 + day;
+                    if (year % 4 == 0)
+                        amountThisYear++;
+                    break;
+                case 6:
+                    amountThisYear = 181 + day;
+                    if (year % 4 == 0)
+                        amountThisYear++;
+                    break;
+                case 7:
+                    amountThisYear = 212 + day;
+                    if (year % 4 == 0)
+                        amountThisYear++;
+                    break;
+                case 8:
+                    amountThisYear = 243 + day;
+                    if (year % 4 == 0)
+                        amountThisYear++;
+                    break;
+                case 9:
+                    amountThisYear = 273 + day;
+                    if (year % 4 == 0)
+                        amountThisYear++;
+                    break;
+                case 10:
+                    amountThisYear = 304 + day;
+                    if (year % 4 == 0)
+                        amountThisYear++;
+                    break;
+                case 11:
+                    amountThisYear = 334 + day;
+                    if (year % 4 == 0)
+                        amountThisYear++;
+                    break;
+            }
+            return (amountPastYears + amountThisYear);
```

Comment:
How about creating one more function, which returns a number of days in month:
```
static int GetNumberOfDays(int month)
{
}
```
You could use it in both `CheckDate` and `CalculateAmountOfDays` methods. 
This `273`, `304`, `334` scare me :) 
I would create a loop for 1 to `month` and sum days in each month one by one